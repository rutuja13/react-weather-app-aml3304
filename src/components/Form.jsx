import { makeStyles } from "@material-ui/core";
import React from "react";
import { useEffect, useState } from "react";
import {getData} from "../service/api"
import Information from "./Information"

const useStyles = makeStyles({
    input:{
        color : '#fff' , 
        marginRight : 15 , 

    } , 
    component :{
        padding : 10 , 
        background : '#445a6f'
    },
    button :{
        width : 150,
        height : 40,
        background : '#e67e22',
        color : '#fff',
        marginTop :5
    }
})

const Form = () =>{
    const classes = useStyles();
    const [data , getWeatherData ] = useState();
    const [city ] = useState("Toronto ");
    const [country ] = useState("CA");
    const [click,handleClick] = useState(true);
    

    useEffect(()=>{
        
        const getWeather = async() =>{
            city &&     await getData(city,country).then(response =>{
                getWeatherData(response.data)
                console.log(response.data)
            })
        }
        getWeather();
        handleClick(true);
    },[click]);
    
    return(
        <React.Fragment>
            
            <Information data = {data}>

            </Information>
        </React.Fragment>
        
    )
}
export default Form;