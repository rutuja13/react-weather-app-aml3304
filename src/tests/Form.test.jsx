import { configure, shallow } from "enzyme";
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Form from "../components/Form";
import * as apiService from "../service/api";


configure({ adapter: new Adapter() });

jest.mock('../service/api', () => {
    return {
        getData: jest.fn()
    };
});

describe('Form Component', () => {
    it('Snapshot Matching', () => {
        const wrapper = shallow(<Form />);
        expect(wrapper.getElements()).toMatchSnapshot();
    });

});

