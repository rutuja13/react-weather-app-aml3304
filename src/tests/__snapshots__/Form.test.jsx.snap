// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component Snapshot Matching 1`] = `
Array [
  <React.Fragment>
    <Information />
  </React.Fragment>,
]
`;
